autoload/vsnip/source/user_snippet.vim

```diff
@@ -23,12 +23,23 @@ function! vsnip#source#user_snippet#refresh(path) abort
   endif
 endfunction
 
+function! s:get_source_dirs(bufnr) abort
+  let l:dirs = [g:vsnip_snippet_dir]
+  let l:dirs += g:vsnip_snippet_dirs
+  let l:dirs += getbufvar(a:bufnr, 'vsnip_snippet_dirs', [])
+  let l:buf_dir = getbufvar(a:bufnr, 'vsnip_snippet_dir', v:null)
```

Comment:
I think we should prefer buffer specific directories so we should change the order of the dirs in here.

Reply:
Done